### ISSUE TITLE
osrf/ros2 docker for ros2 doesn't work

### ISSUE THREAD

**Issue by claireyywang** [osrf/docker_images#358]
https://hub.docker.com/r/osrf/ros2/ this is the first search result on google when I searched for `docker ros2` and the command `docker pull osrf/ros2` does not work. 
```
Using default tag: latest
Error response from daemon: manifest for osrf/ros2:latest not found: manifest unknown: manifest unknown
```
This page is misleading. I believe the actual docker page for all ros related images is this one https://hub.docker.com/_/ros . There have been four similar issues on this matter, which I think proves the amount of traffic on the site. Can we either update the pull command, or add information on the page to redirect users to where they are supposed to go?

**Comment by mikaelarguedas** [comment]
Closing as duplicate of https://github.com/osrf/docker_images/issues/186

**Comment by mikaelarguedas** [comment]
Sorry for the premature closing I read the description too fast.

The rationale for not having a latest tag is highlighted on #186. 

> Can we either update the pull command,

Unfortunately I'm not aware of a way to change the command shown for pulling.

> or add information on the page to redirect users to where they are supposed to go?

The header of the webpage is "**Experimental**, Docker images for ROS 2 development only"
And below is a link to the ROS official images.
I added "No `latest` tag, pull with *explicit* tags" to the header but couldn't come up with something clearer given the description's 100-character limit.

Do you have a specific suggestion or wording in mind that could make it clearer for users landing on the page ?

**Comment by claireyywang** [comment]
> The header of the webpage is "**Experimental**, Docker images for ROS 2 development only"
> And below is a link to the ROS official images.

To be honest the `ROS` link didn't even stand out to me as a link. I had to go back and look for it after reading your comment. And to someone not familiar with docker but just want to get ros2 up and running, it requires quite a bit of reading to tell which one of the two sites is supposed to work. 

> Do you have a specific suggestion or wording in mind that could make it clearer for users landing on the page ?

Maybe edit the header or the description (very top underneath `OSRF Docker Images`) to say something like 

" This repo contains images specific to ROS 2, and are particularly for development and continuous integration. These may contain experimental features or unstable interfaces for testing the latest ROS 2 branches. Go to  https://hub.docker.com/_/ros for official stable releases of ROS and ROS 2 images. "

Just to give people an idea of what this page has and how it is different from  https://hub.docker.com/_/ros page, and that would help them decide which one they want. 

Then for people who specifically want this image, we can say "the pull command doesn't work as is, but requires specific tags for example `docker pull osrf/ros2: nightly`, see more tags below" (cuz I didn't even know the default tag is `latest` when I first started).

All those may sound trivial to experienced docker users, but for beginners (like me), they are super helpful to know.

**Comment by mikaelarguedas** [comment]
TLDR; 
We need to modify the README of this repo to change the content of the DockerHub pages (more details of the implications further below).

One way to help the confusion is to add a table of content at the top of the README.
This allows to point to the right paragraph in the README without changing the semantic of the page, or redirecting people to other pages without pointing them to the paragraph describing the repo first. 
For example: https://github.com/osrf/docker_images/blob/8046b09e85567da6a10db4aa3104dba04f59089e/README.md

Would that have helped you find the right resource? 

---

> Maybe edit the header or the description (very top underneath OSRF Docker Images) to say something like

Unfortunately content outside of the header cannot be `osrf/ros2` specific ATM.
The only part of the page that is hosted on Dockerhub is the 100-character description at the top. Everything else is [the README of this Github repository](https://github.com/osrf/docker_images/blob/master/README.md), that is not specific to ROS 2 or experimental images. This is where all the Dockerfiles (ROS1, ROS2, Gazebo) populating both the official library and the osrf profile are.

This is the same one used for
https://hub.docker.com/r/osrf/ros/
https://hub.docker.com/r/osrf/ros2/
https://hub.docker.com/r/osrf/gazebo/
https://hub.docker.com/r/osrf/ros_legacy
and the description of https://github.com/osrf/docker_images

So any change targeting something else than the osrf/ros2 Dockerhub header description would need to fit in that context.


> All those may sound trivial to experienced docker users, but for beginners (like me), they are super helpful to know.

Agreed that we should try to limit confusion for users landing on that page. However I believe it is ok to assume some knowledge of the tools (what is an image, container, tag, how to pull a tag etc). The same way as ROS documentation is not trying to teach people how to Linux, bash or C++.


>  this is the first search result on google when I searched for docker ros2 

Interesting, I tried similar searches but always land on the official dockerhub page:
https://duckduckgo.com/?q=docker+ros2&t=h_&ia=software
https://duckduckgo.com/?q=ros2+docker+image&t=h_&ia=web
https://duckduckgo.com/?q=ros+eloquent+docker+image&t=h_&ia=web

Google's targeted results..

**Comment by mikaelarguedas** [comment]
@claireyywang I opened #363, could you please have a look ?

**Comment by claireyywang** [comment]
Sorry about the late reply. I think the table of content helps a lot in navigating different resources and is a good temporary solution. Thanks for explaining and addressing this issue. Feel free to close it anytime.

**Comment by mikaelarguedas** [comment]
Addressed in #363
